<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Ju Yeong Kim" />

<meta name="date" content="2021-08-30" />

<title>Introduction to DataSpaceR</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Introduction to DataSpaceR</h1>
<h4 class="author">Ju Yeong Kim</h4>
<h4 class="date">2021-08-30</h4>



<style>
table {
    display: block;
    max-width: 100%;
    max-height: 600px;
    overflow: scroll;
}
thead{
    position: sticky;
    top: 0px;
    background-color: #fff;
}
</style>
<p>This package provides a thin wrapper around <a href="https://cran.r-project.org/package=Rlabkey">Rlabkey</a> and connects to the the <a href="https://dataspace.cavd.org">CAVD DataSpace</a> database, making it easier to fetch datasets from specific studies.</p>
<div id="configuration" class="section level2">
<h2>Configuration</h2>
<p>First, go to <a href="https://dataspace.cavd.org">DataSpace</a> now and set yourself up with an account.</p>
<p>In order to connect to the CAVD DataSpace via <code>DataSpaceR</code>, you will need a <code>netrc</code> file in your home directory that will contain a <code>machine</code> name (hostname of DataSpace), and <code>login</code> and <code>password</code>. There are two ways to create a <code>netrc</code> file.</p>
<div id="creating-a-netrc-file-with-writenetrc" class="section level3">
<h3>Creating a netrc file with <code>writeNetrc</code></h3>
<p>On your R console, create a <code>netrc</code> file using a function from <code>DataSpaceR</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">writeNetrc</span>(</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">login =</span> <span class="st">&quot;yourEmail@address.com&quot;</span>,</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">password =</span> <span class="st">&quot;yourSecretPassword&quot;</span>,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">netrcFile =</span> <span class="st">&quot;/your/home/directory/.netrc&quot;</span> <span class="co"># use getNetrcPath() to get the default path</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p>This will create a <code>netrc</code> file in your home directory. Make sure you have a valid login and password.</p>
</div>
<div id="manually-creating-a-netrc-file" class="section level3">
<h3>Manually creating a netrc file</h3>
<p><strong><em>Alternatively</em></strong>, you can manually create a netrc file.</p>
<ul>
<li>On Windows, this file should be named <code>_netrc</code></li>
<li>On UNIX/Mac, it should be named <code>.netrc</code></li>
<li>The file should be located in the user’s home directory, and the permissions on the file should be unreadable for everybody except the owner</li>
<li>To determine your home directory, run <code>Sys.getenv(&quot;HOME&quot;)</code> in R</li>
</ul>
<p>The following three lines must be included in the <code>.netrc</code> or <code>_netrc</code> file either separated by white space (spaces, tabs, or newlines) or commas. Multiple such blocks can exist in one file.</p>
<pre><code>machine dataspace.cavd.org
login myuser@domain.com
password supersecretpassword</code></pre>
<p>See <a href="https://www.labkey.org/wiki/home/Documentation/page.view?name=netrc">here</a> for more information about <code>netrc</code>.</p>
</div>
</div>
<div id="initiate-a-connection" class="section level2">
<h2>Initiate a connection</h2>
<p>We’ll be looking at study <code>cvd256</code>. If you want to use a different study, change that string. You can instantiate multiple connections to different studies simultaneously.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DataSpaceR)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>con <span class="ot">&lt;-</span> <span class="fu">connectDS</span>()</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>con</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;DataSpaceConnection&gt;</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   URL: https://dataspace.cavd.org</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   User: </span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available studies: 273</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - 77 studies with data</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - 5049 subjects</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - 423195 data points</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available groups: 3</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available publications: 1530</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - 12 publications with data</span></span></code></pre></div>
<p>The call to <code>connectDS</code> instantiates the connection. Printing the object shows where it’s connected and the available studies.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(con<span class="sc">$</span>availableStudies))</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="2%" />
<col width="6%" />
<col width="56%" />
<col width="4%" />
<col width="1%" />
<col width="3%" />
<col width="4%" />
<col width="2%" />
<col width="8%" />
<col width="1%" />
<col width="3%" />
<col width="4%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">study_name</th>
<th align="left">short_name</th>
<th align="left">title</th>
<th align="left">type</th>
<th align="left">status</th>
<th align="left">stage</th>
<th align="left">species</th>
<th align="left">start_date</th>
<th align="left">strategy</th>
<th align="left">network</th>
<th align="left">data_availability</th>
<th align="left">ni_data_availability</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cor01</td>
<td align="left">NA</td>
<td align="left">The correlate of risk targeted intervention study (CORTIS): A randomized, partially-blinded, clinical trial of isoniazid and rifapentine (3HP) therapy to prevent pulmonary tuberculosis in high-risk individuals identified by a transcriptomic correlate of risk</td>
<td align="left">Phase III</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Human</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">GH-VAP</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">cvd232</td>
<td align="left">Parks_RV_232</td>
<td align="left">​Limiting Dose Vaginal SIVmac239 Challenge of RhCMV-SIV vaccinated Indian rhesus macaques.</td>
<td align="left">Pre-Clinical NHP</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Rhesus macaque</td>
<td align="left">2009-11-24</td>
<td align="left">Vector vaccines (viral or bacterial)</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">cvd234</td>
<td align="left">Zolla-Pazner_Mab_test1 Study</td>
<td align="left">Zolla-Pazner_Mab_Test1</td>
<td align="left">Antibody Screening</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Non-Organism Study</td>
<td align="left">2009-02-03</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">cvd235</td>
<td align="left">mAbs potency</td>
<td align="left">Weiss mAbs potency</td>
<td align="left">Antibody Screening</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Non-Organism Study</td>
<td align="left">2008-08-21</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">cvd236</td>
<td align="left">neutralization assays</td>
<td align="left">neutralization assays</td>
<td align="left">Antibody Screening</td>
<td align="left">Active</td>
<td align="left">In Progress</td>
<td align="left">Non-Organism Study</td>
<td align="left">2009-02-03</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">cvd238</td>
<td align="left">Gallo_PA_238</td>
<td align="left">HIV-1 neutralization responses in chronically infected individuals</td>
<td align="left">Antibody Screening</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Non-Organism Study</td>
<td align="left">2009-01-08</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<p><code>con$availableStudies</code> shows the available studies in the CAVD DataSpace. Check out <a href="https://docs.ropensci.org/DataSpaceR/reference/DataSpaceConnection.html">the reference page</a> of <code>DataSpaceConnection</code> for all available fields and methods.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>cvd256 <span class="ot">&lt;-</span> con<span class="sc">$</span><span class="fu">getStudy</span>(<span class="st">&quot;cvd256&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>cvd256</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;DataSpaceStudy&gt;</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Study: cvd256</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   URL: https://dataspace.cavd.org/CAVD/cvd256</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available datasets:</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Binding Ab multiplex assay</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Demographics</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Neutralizing antibody</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available non-integrated datasets:</span></span></code></pre></div>
<p><code>con$getStudy</code> creates a connection to the study <code>cvd256</code>. Printing the object shows where it’s connected, to what study, and the available datasets.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(cvd256<span class="sc">$</span>availableDatasets)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">name</th>
<th align="left">label</th>
<th align="right">n</th>
<th align="left">integrated</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">BAMA</td>
<td align="left">Binding Ab multiplex assay</td>
<td align="right">6740</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Demographics</td>
<td align="left">Demographics</td>
<td align="right">121</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NAb</td>
<td align="left">Neutralizing antibody</td>
<td align="right">1419</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(cvd256<span class="sc">$</span>treatmentArm)</span></code></pre></div>
<table>
<colgroup>
<col width="7%" />
<col width="4%" />
<col width="5%" />
<col width="4%" />
<col width="7%" />
<col width="8%" />
<col width="4%" />
<col width="55%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">arm_id</th>
<th align="left">arm_part</th>
<th align="left">arm_group</th>
<th align="left">arm_name</th>
<th align="left">randomization</th>
<th align="left">coded_label</th>
<th align="right">last_day</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cvd256-NA-A-A</td>
<td align="left">NA</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">Vaccine</td>
<td align="left">Group A Vaccine</td>
<td align="right">168</td>
<td align="left">DNA-C 4 mg administered IM at weeks 0, 4, and 8 AND NYVAC-C 10^7pfu/mL administered IM at week 24</td>
</tr>
<tr class="even">
<td align="left">cvd256-NA-B-B</td>
<td align="left">NA</td>
<td align="left">B</td>
<td align="left">B</td>
<td align="left">Vaccine</td>
<td align="left">Group B Vaccine</td>
<td align="right">168</td>
<td align="left">DNA-C 4 mg administered IM at weeks 0 and 4 AND NYVAC-C 10^7pfu/mL administered IM at weeks 20 and 24</td>
</tr>
</tbody>
</table>
<p>Available datasets and treatment arm information for the connection can be accessed by <code>availableDatasets</code> and <code>treatmentArm</code>.</p>
</div>
<div id="fetching-datasets" class="section level2">
<h2>Fetching datasets</h2>
<p>We can grab any of the datasets listed in the connection (<code>availableDatasets</code>).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>NAb <span class="ot">&lt;-</span> cvd256<span class="sc">$</span><span class="fu">getDataset</span>(<span class="st">&quot;NAb&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(NAb)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 1419   33</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(NAb)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;participant_id&quot;      &quot;participant_visit&quot;   &quot;visit_day&quot;          </span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [4] &quot;assay_identifier&quot;    &quot;summary_level&quot;       &quot;specimen_type&quot;      </span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;antigen&quot;             &quot;antigen_type&quot;        &quot;virus&quot;              </span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [10] &quot;virus_type&quot;          &quot;virus_insert_name&quot;   &quot;clade&quot;              </span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] &quot;neutralization_tier&quot; &quot;tier_clade_virus&quot;    &quot;target_cell&quot;        </span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [16] &quot;initial_dilution&quot;    &quot;titer_ic50&quot;          &quot;titer_ic80&quot;         </span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] &quot;response_call&quot;       &quot;nab_lab_source_key&quot;  &quot;lab_code&quot;           </span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [22] &quot;exp_assayid&quot;         &quot;titer_id50&quot;          &quot;titer_id80&quot;         </span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] &quot;nab_response_id50&quot;   &quot;nab_response_id80&quot;   &quot;slope&quot;              </span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [28] &quot;vaccine_matched&quot;     &quot;study_prot&quot;          &quot;virus_full_name&quot;    </span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] &quot;virus_species&quot;       &quot;virus_host_cell&quot;     &quot;virus_backbone&quot;</span></span></code></pre></div>
<p>The <em>cvd256</em> object is an <a href="https://cran.r-project.org/package=R6"><code>R6</code></a> class, so it behaves like a true object. Functions (like <code>getDataset</code>) are members of the object, thus the <code>$</code> semantics to access member functions.</p>
<p>We can get detailed variable information using <code>getDatasetDescription</code>. <code>getDataset</code> and <code>getDatasetDescription</code> accept either the <code>name</code> or <code>label</code> field listed in <code>availableDatasets</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(cvd256<span class="sc">$</span><span class="fu">getDatasetDescription</span>(<span class="st">&quot;NAb&quot;</span>))</span></code></pre></div>
<table>
<colgroup>
<col width="7%" />
<col width="17%" />
<col width="8%" />
<col width="66%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">fieldName</th>
<th align="left">caption</th>
<th align="left">type</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ParticipantId</td>
<td align="left">Participant ID</td>
<td align="left">Text (String)</td>
<td align="left">Subject identifier</td>
</tr>
<tr class="even">
<td align="left">antigen</td>
<td align="left">Antigen name</td>
<td align="left">Text (String)</td>
<td align="left">The name of the antigen (virus) being tested.</td>
</tr>
<tr class="odd">
<td align="left">antigen_type</td>
<td align="left">Antigen type</td>
<td align="left">Text (String)</td>
<td align="left">The standardized term for the type of virus used in the construction of the nAb antigen.</td>
</tr>
<tr class="even">
<td align="left">assay_identifier</td>
<td align="left">Assay identifier</td>
<td align="left">Text (String)</td>
<td align="left">Name identifying assay</td>
</tr>
<tr class="odd">
<td align="left">clade</td>
<td align="left">Virus clade</td>
<td align="left">Text (String)</td>
<td align="left">The clade (gene subtype) of the virus (antigen) being tested.</td>
</tr>
<tr class="even">
<td align="left">exp_assayid</td>
<td align="left">Experimental Assay Design Code</td>
<td align="left">Integer</td>
<td align="left">Unique ID assigned to the experiment design of the assay for tracking purposes.</td>
</tr>
<tr class="odd">
<td align="left">initial_dilution</td>
<td align="left">Initial dilution</td>
<td align="left">Number (Double)</td>
<td align="left">Indicates the initial specimen dilution.</td>
</tr>
<tr class="even">
<td align="left">lab_code</td>
<td align="left">Lab ID</td>
<td align="left">Text (String)</td>
<td align="left">A code indicating the lab performing the assay.</td>
</tr>
<tr class="odd">
<td align="left">nab_lab_source_key</td>
<td align="left">Data provenance</td>
<td align="left">Integer</td>
<td align="left">Details regarding the provenance of the assay results.</td>
</tr>
<tr class="even">
<td align="left">nab_response_ID50</td>
<td align="left">Response call ID50</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if neutralization is detected based on ID50 titer.</td>
</tr>
<tr class="odd">
<td align="left">nab_response_ID80</td>
<td align="left">Response call ID80</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if neutralization is detected based on ID80 titer.</td>
</tr>
<tr class="even">
<td align="left">neutralization_tier</td>
<td align="left">Neutralization tier</td>
<td align="left">Text (String)</td>
<td align="left">A classification specific to HIV NAb assay design, in which an antigen is assessed for its ease of neutralization (1=most easily neutralized, 3=least easily neutralized)</td>
</tr>
<tr class="odd">
<td align="left">response_call</td>
<td align="left">Response call</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if neutralization is detected.</td>
</tr>
<tr class="even">
<td align="left">slope</td>
<td align="left">Slope</td>
<td align="left">Number (Double)</td>
<td align="left">The slope calculated using the difference between 50% and 80% neutralization.</td>
</tr>
<tr class="odd">
<td align="left">specimen_type</td>
<td align="left">Specimen type</td>
<td align="left">Text (String)</td>
<td align="left">The type of specimen used in the assay. For nAb assays, this is generally serum or plasma.</td>
</tr>
<tr class="even">
<td align="left">study_prot</td>
<td align="left">Study Protocol</td>
<td align="left">Text (String)</td>
<td align="left">Study protocol</td>
</tr>
<tr class="odd">
<td align="left">summary_level</td>
<td align="left">Data summary level</td>
<td align="left">Text (String)</td>
<td align="left">Defines the level at which the magnitude or response has been summarized (e.g. summarized at the isolate level).</td>
</tr>
<tr class="even">
<td align="left">target_cell</td>
<td align="left">Target cell</td>
<td align="left">Text (String)</td>
<td align="left">The cell line used in the assay to determine infection (lack of neutralization). Generally TZM-bl or A3R5, but can also be other cell lines or non-engineered cells.</td>
</tr>
<tr class="odd">
<td align="left">tier_clade_virus</td>
<td align="left">Neutralization tier + Antigen clade + Virus</td>
<td align="left">Text (String)</td>
<td align="left">A combination of neutralization tier, antigen clade, and virus used for filtering.</td>
</tr>
<tr class="even">
<td align="left">titer_ID50</td>
<td align="left">Titer ID50</td>
<td align="left">Number (Double)</td>
<td align="left">The adjusted value of 50% maximal inhibitory dilution (ID50).</td>
</tr>
<tr class="odd">
<td align="left">titer_ID80</td>
<td align="left">Titer ID80</td>
<td align="left">Number (Double)</td>
<td align="left">The adjusted value of 80% maximal inhibitory dilution (ID80).</td>
</tr>
<tr class="even">
<td align="left">titer_ic50</td>
<td align="left">Titer IC50</td>
<td align="left">Number (Double)</td>
<td align="left">The half maximal inhibitory concentration (IC50).</td>
</tr>
<tr class="odd">
<td align="left">titer_ic80</td>
<td align="left">Titer IC80</td>
<td align="left">Number (Double)</td>
<td align="left">The 80% maximal inhibitory concentration (IC80).</td>
</tr>
<tr class="even">
<td align="left">vaccine_matched</td>
<td align="left">Antigen vaccine match indicator</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if the interactive part of the antigen was designed to match the immunogen in the vaccine.</td>
</tr>
<tr class="odd">
<td align="left">virus</td>
<td align="left">Virus name</td>
<td align="left">Text (String)</td>
<td align="left">The term for the virus (antigen) being tested.</td>
</tr>
<tr class="even">
<td align="left">virus_backbone</td>
<td align="left">Virus backbone</td>
<td align="left">Text (String)</td>
<td align="left">Indicates the backbone used to generate the virus if from a different plasmid than the envelope.</td>
</tr>
<tr class="odd">
<td align="left">virus_full_name</td>
<td align="left">Virus full name</td>
<td align="left">Text (String)</td>
<td align="left">The full name of the virus used in the construction of the nAb antigen.</td>
</tr>
<tr class="even">
<td align="left">virus_host_cell</td>
<td align="left">Virus host cell</td>
<td align="left">Text (String)</td>
<td align="left">The host cell used to incubate the virus stock.</td>
</tr>
<tr class="odd">
<td align="left">virus_insert_name</td>
<td align="left">Virus insert name</td>
<td align="left">Text (String)</td>
<td align="left">The amino acid sequence inserted in the virus construct.</td>
</tr>
<tr class="even">
<td align="left">virus_species</td>
<td align="left">Virus species</td>
<td align="left">Text (String)</td>
<td align="left">A classification for virus species using informal taxonomy.</td>
</tr>
<tr class="odd">
<td align="left">virus_type</td>
<td align="left">Virus type</td>
<td align="left">Text (String)</td>
<td align="left">The type of virus used in the construction of the nAb antigen.</td>
</tr>
<tr class="even">
<td align="left">visit_day</td>
<td align="left">Visit Day</td>
<td align="left">Integer</td>
<td align="left">Target study day defined for a study visit. Study days are relative to Day 0, where Day 0 is typically defined as enrollment and/or first injection.</td>
</tr>
</tbody>
</table>
<p>To get only a subset of the data and speed up the download, filters can be passed to <code>getDataset</code>. The filters are created using the <code>makeFilter</code> function of the <code>Rlabkey</code> package.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>cvd256Filter <span class="ot">&lt;-</span> <span class="fu">makeFilter</span>(<span class="fu">c</span>(<span class="st">&quot;visit_day&quot;</span>, <span class="st">&quot;EQUAL&quot;</span>, <span class="st">&quot;0&quot;</span>))</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>NAb_day0 <span class="ot">&lt;-</span> cvd256<span class="sc">$</span><span class="fu">getDataset</span>(<span class="st">&quot;NAb&quot;</span>, <span class="at">colFilter =</span> cvd256Filter)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(NAb_day0)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 709  33</span></span></code></pre></div>
<p>See <code>?makeFilter</code> for more information on the syntax.</p>
</div>
<div id="creating-a-connection-to-all-studies" class="section level2">
<h2>Creating a connection to all studies</h2>
<p>To fetch data from multiple studies, create a connection at the project level.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>cavd <span class="ot">&lt;-</span> con<span class="sc">$</span><span class="fu">getStudy</span>(<span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>This will instantiate a connection at the <code>CAVD</code> level. Most functions work cross study connections just like they do on single studies.</p>
<p>You can get a list of datasets available across all studies.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>cavd</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;DataSpaceStudy&gt;</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Study: CAVD</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   URL: https://dataspace.cavd.org/CAVD</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available datasets:</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Binding Ab multiplex assay</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Demographics</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Enzyme-Linked ImmunoSpot</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Intracellular Cytokine Staining</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Neutralizing antibody</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - PK MAb</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available non-integrated datasets:</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(cavd<span class="sc">$</span>availableDatasets)</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">name</th>
<th align="left">label</th>
<th align="right">n</th>
<th align="left">integrated</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">BAMA</td>
<td align="left">Binding Ab multiplex assay</td>
<td align="right">170320</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Demographics</td>
<td align="left">Demographics</td>
<td align="right">5049</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">ELISPOT</td>
<td align="left">Enzyme-Linked ImmunoSpot</td>
<td align="right">5610</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ICS</td>
<td align="left">Intracellular Cytokine Staining</td>
<td align="right">195883</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NAb</td>
<td align="left">Neutralizing antibody</td>
<td align="right">51382</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">PKMAb</td>
<td align="left">PK MAb</td>
<td align="right">3217</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<p>In all-study connection, <code>getDataset</code> will combine the requested datasets. Note that in most cases, the datasets will have too many subjects for quick data transfer, making filtering of the data a necessity. The <code>colFilter</code> argument can be used here, as described in the <code>getDataset</code> section.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>conFilter <span class="ot">&lt;-</span> <span class="fu">makeFilter</span>(<span class="fu">c</span>(<span class="st">&quot;species&quot;</span>, <span class="st">&quot;EQUAL&quot;</span>, <span class="st">&quot;Human&quot;</span>))</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>human <span class="ot">&lt;-</span> cavd<span class="sc">$</span><span class="fu">getDataset</span>(<span class="st">&quot;Demographics&quot;</span>, <span class="at">colFilter =</span> conFilter)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(human)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 3142   36</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(human)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] &quot;subject_id&quot;                      &quot;subject_visit&quot;                  </span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] &quot;species&quot;                         &quot;subspecies&quot;                     </span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] &quot;sexatbirth&quot;                      &quot;race&quot;                           </span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] &quot;ethnicity&quot;                       &quot;country_enrollment&quot;             </span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] &quot;circumcised_enrollment&quot;          &quot;bmi_enrollment&quot;                 </span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] &quot;agegroup_range&quot;                  &quot;agegroup_enrollment&quot;            </span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] &quot;age_enrollment&quot;                  &quot;study_label&quot;                    </span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [15] &quot;study_start_date&quot;                &quot;study_first_enr_date&quot;           </span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [17] &quot;study_fu_complete_date&quot;          &quot;study_public_date&quot;              </span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [19] &quot;study_network&quot;                   &quot;study_last_vaccination_day&quot;     </span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [21] &quot;study_type&quot;                      &quot;study_part&quot;                     </span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [23] &quot;study_group&quot;                     &quot;study_arm&quot;                      </span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [25] &quot;study_arm_summary&quot;               &quot;study_arm_coded_label&quot;          </span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [27] &quot;study_randomization&quot;             &quot;study_product_class_combination&quot;</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [29] &quot;study_product_combination&quot;       &quot;study_short_name&quot;               </span></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [31] &quot;study_grant_pi_name&quot;             &quot;study_strategy&quot;                 </span></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [33] &quot;study_prot&quot;                      &quot;genderidentity&quot;                 </span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [35] &quot;studycohort&quot;                     &quot;bmi_category&quot;</span></span></code></pre></div>
<p>Check out <a href="https://docs.ropensci.org/DataSpaceR/reference/DataSpaceStudy.html">the reference page</a> of <code>DataSpaceStudy</code> for all available fields and methods.</p>
</div>
<div id="connect-to-a-saved-group" class="section level2">
<h2>Connect to a saved group</h2>
<p>A group is a curated collection of participants from filtering of treatments, products, studies, or species, and it is created in <a href="https://dataspace.cavd.org/cds/CAVD/app.view">the DataSpace App</a>.</p>
<p>Let’s say you are using the App to filter and visualize data and want to save them for later or explore in R with <code>DataSpaceR</code>. You can save a group by clicking the Save button on the Active Filter Panel.</p>
<p>We can browse available the saved groups or the curated groups by DataSpace Team via <code>availableGroups</code>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(con<span class="sc">$</span>availableGroups)</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="3%" />
<col width="13%" />
<col width="13%" />
<col width="48%" />
<col width="4%" />
<col width="2%" />
<col width="1%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="right">group_id</th>
<th align="left">label</th>
<th align="left">original_label</th>
<th align="left">description</th>
<th align="left">created_by</th>
<th align="left">shared</th>
<th align="right">n</th>
<th align="left">studies</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">220</td>
<td align="left">NYVAC durability comparison</td>
<td align="left">NYVAC_durability</td>
<td align="left">Compare durability in 4 NHP studies using NYVAC-C (vP2010) and NYVAC-KC-gp140 (ZM96) products.</td>
<td align="left">ehenrich</td>
<td align="left">TRUE</td>
<td align="right">78</td>
<td align="left">cvd281, cvd434, cvd259, cvd277</td>
</tr>
<tr class="even">
<td align="right">228</td>
<td align="left">HVTN 505 case control subjects</td>
<td align="left">HVTN 505 case control subjects</td>
<td align="left">Participants from HVTN 505 included in the case-control analysis</td>
<td align="left">drienna</td>
<td align="left">TRUE</td>
<td align="right">189</td>
<td align="left">vtn505</td>
</tr>
<tr class="odd">
<td align="right">230</td>
<td align="left">HVTN 505 polyfunctionality vs BAMA</td>
<td align="left">HVTN 505 polyfunctionality vs BAMA</td>
<td align="left">Compares ICS polyfunctionality (CD8+, Any Env) to BAMA mfi-delta (single Env antigen) in the HVTN 505 case control cohort</td>
<td align="left">drienna</td>
<td align="left">TRUE</td>
<td align="right">170</td>
<td align="left">vtn505</td>
</tr>
</tbody>
</table>
<p>To fetch data from a saved group, create a connection at the project level with a group ID. For example, we can connect to the “NYVAC durability comparison” group which has group ID 220 by <code>getGroup</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>nyvac <span class="ot">&lt;-</span> con<span class="sc">$</span><span class="fu">getGroup</span>(<span class="dv">220</span>)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>nyvac</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &lt;DataSpaceStudy&gt;</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Group: NYVAC durability comparison</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   URL: https://dataspace.cavd.org/CAVD</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available datasets:</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Binding Ab multiplex assay</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Demographics</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Enzyme-Linked ImmunoSpot</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Intracellular Cytokine Staining</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     - Neutralizing antibody</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Available non-integrated datasets:</span></span></code></pre></div>
<p>Retrieving a dataset is the same as before.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>NAb_nyvac <span class="ot">&lt;-</span> nyvac<span class="sc">$</span><span class="fu">getDataset</span>(<span class="st">&quot;NAb&quot;</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(NAb_nyvac)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 4281   33</span></span></code></pre></div>
</div>
<div id="access-virus-metadata" class="section level2">
<h2>Access Virus Metadata</h2>
<p>DataSpace maintains metadata about all viruses used in Neutralizing Antibody (NAb) assays. This data can be accessed through the app on the <a href="https://dataspace.cavd.org/cds/CAVD/app.view#learn/learn/Assay/NAB/antigens">NAb antigen page</a> and <a href="https://dataspace.cavd.org/cds/CAVD/app.view#learn/learn/Assay/NAB%20MAB/antigens">NAb MAb antigen page</a>.</p>
<p>We can access this metadata in <code>DataSpaceR</code> with <code>con$virusMetadata</code>:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(con<span class="sc">$</span>virusMetadata))</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="8%" />
<col width="6%" />
<col width="7%" />
<col width="9%" />
<col width="2%" />
<col width="7%" />
<col width="15%" />
<col width="8%" />
<col width="6%" />
<col width="7%" />
<col width="7%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">assay_identifier</th>
<th align="left">virus</th>
<th align="left">virus_type</th>
<th align="left">neutralization_tier</th>
<th align="left">clade</th>
<th align="left">antigen_control</th>
<th align="left">virus_full_name</th>
<th align="left">virus_name_other</th>
<th align="left">virus_species</th>
<th align="left">virus_host_cell</th>
<th align="left">virus_backbone</th>
<th align="left">panel_names</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">NAB MAB</td>
<td align="left">0013095-2.11</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">NA</td>
<td align="left">0</td>
<td align="left">0013095-2.11 [SG3Δenv] 293T/17</td>
<td align="left">NA</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
<tr class="even">
<td align="left">NAB MAB</td>
<td align="left">001428-2.42</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">001428-2.42 [SG3Δenv] 293T/17</td>
<td align="left">NA</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
<tr class="odd">
<td align="left">NAB MAB</td>
<td align="left">0041.v3.c18</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">0041.v3.c18 [SG3Δenv] 293T/17</td>
<td align="left">0041.V3.C18</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NAB MAB</td>
<td align="left">0077.v1.c16</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">0077.v1.c16 [SG3Δenv] 293T/17</td>
<td align="left">0077.v1.c16</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NAB</td>
<td align="left">00836-2.5</td>
<td align="left">Env Pseudotype</td>
<td align="left">1B</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">00836-2.5 [SG3Δenv] 293T/17</td>
<td align="left">NA</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
<tr class="even">
<td align="left">NAB MAB</td>
<td align="left">0260.v5.c1</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">A</td>
<td align="left">0</td>
<td align="left">0260.v5.c1 [SG3Δenv] 293T/17</td>
<td align="left">0260.V5.C1</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
</tbody>
</table>
</div>
<div id="access-monoclonal-antibody-data" class="section level2">
<h2>Access monoclonal antibody data</h2>
<p>See other vignette for a tutorial on accessing monoclonal antibody data with <code>DataSpaceR</code>:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vignette</span>(<span class="st">&quot;Monoconal_Antibody_Data&quot;</span>)</span></code></pre></div>
</div>
<div id="browse-and-download-publication-data" class="section level2">
<h2>Browse and Download Publication Data</h2>
<p>DataSpace maintains a curated collection of relevant publications, which can be accessed through the <a href="https://dataspace.cavd.org/cds/CAVD/app.view?#learn/learn/Publication">Publications page</a> through the app. Metadata about these publications can be accessed through <code>DataSpaceR</code> with <code>con$availablePublications</code>.</p>
<p>See Publication Data vignette for a tutorial on accessing publication data through DataSpaceR.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">vignette</span>(<span class="st">&quot;Publication_Data&quot;</span>)</span></code></pre></div>
</div>
<div id="reference-tables" class="section level2">
<h2>Reference Tables</h2>
<p>The followings are the tables of all fields and methods that work on <code>DataSpaceConnection</code> and <code>DataSpaceStudy</code> objects and could be used as a quick reference.</p>
<div id="dataspaceconnection" class="section level3">
<h3><code>DataSpaceConnection</code></h3>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>availableStudies</code></td>
<td>The table of available studies.</td>
</tr>
<tr class="even">
<td><code>availableGroups</code></td>
<td>The table of available groups.</td>
</tr>
<tr class="odd">
<td><code>availablePublications</code></td>
<td>The table of available publications.</td>
</tr>
<tr class="even">
<td><code>mabGrid</code></td>
<td>The filtered mAb grid.</td>
</tr>
<tr class="odd">
<td><code>mabGridSummary</code></td>
<td>The summarized mAb grid with updated <code>n_</code> columns and <code>geometric_mean_curve_ic50</code>.</td>
</tr>
<tr class="even">
<td><code>virusMetadata</code></td>
<td>Metadata about all viruses in the DataSpace.</td>
</tr>
<tr class="odd">
<td><code>filterMabGrid</code></td>
<td>Filter rows in the mAb grid by specifying the values to keep in the columns found in the <code>mabGrid</code> field.</td>
</tr>
<tr class="even">
<td><code>resetMabGrid</code></td>
<td>Reset the mAb grid to the unfiltered state.</td>
</tr>
<tr class="odd">
<td><code>getMab</code></td>
<td>Create a <code>DataSpaceMab</code> object by filtered <code>mabGrid</code>.</td>
</tr>
<tr class="even">
<td><code>getStudy</code></td>
<td>Create a <code>DataSpaceStudy</code> object by study.</td>
</tr>
<tr class="odd">
<td><code>getGroup</code></td>
<td>Create a <code>DataSpaceStudy</code> object by group.</td>
</tr>
<tr class="even">
<td><code>downloadPublicationData</code></td>
<td>Download data from a chosen publication.</td>
</tr>
</tbody>
</table>
</div>
<div id="dataspacestudy" class="section level3">
<h3><code>DataSpaceStudy</code></h3>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>study</code></td>
<td>The study name.</td>
</tr>
<tr class="even">
<td><code>group</code></td>
<td>The group name.</td>
</tr>
<tr class="odd">
<td><code>availableDatasets</code></td>
<td>The table of datasets available in the study object.</td>
</tr>
<tr class="even">
<td><code>treatmentArm</code></td>
<td>The table of treatment arm information for the connected study. Not available for all study connection.</td>
</tr>
<tr class="odd">
<td><code>dataDir</code></td>
<td>The default target directory for downloading non-integrated datasets.</td>
</tr>
<tr class="even">
<td><code>studyInfo</code></td>
<td>Stores the information about the study.</td>
</tr>
<tr class="odd">
<td><code>getDataset</code></td>
<td>Get a dataset from the connection.</td>
</tr>
<tr class="even">
<td><code>getDatasetDescription</code></td>
<td>Get variable information.</td>
</tr>
<tr class="odd">
<td><code>setDataDir</code></td>
<td>Set default target directory for downloading non-integrated datasets.</td>
</tr>
</tbody>
</table>
</div>
<div id="dataspacemab" class="section level3">
<h3><code>DataSpaceMab</code></h3>
<table>
<colgroup>
<col width="50%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>studyAndMabs</code></td>
<td>The table of available mAbs by study.</td>
</tr>
<tr class="even">
<td><code>mabs</code></td>
<td>The table of available mAbs and their attributes.</td>
</tr>
<tr class="odd">
<td><code>nabMab</code></td>
<td>The table of mAbs and their neutralizing measurements against viruses.</td>
</tr>
<tr class="even">
<td><code>studies</code></td>
<td>The table of available studies.</td>
</tr>
<tr class="odd">
<td><code>assays</code></td>
<td>The table of assay status by study.</td>
</tr>
<tr class="even">
<td><code>variableDefinitions</code></td>
<td>The table of variable definitions.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; R version 4.1.0 (2021-05-18)</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Running under: Ubuntu 20.04.2 LTS</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Matrix products: default</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span></span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; locale:</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attached base packages:</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; other attached packages:</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] data.table_1.14.0 DataSpaceR_0.7.5  knitr_1.33       </span></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [1] Rcpp_1.0.7       digest_0.6.27    assertthat_0.2.1 R6_2.5.0        </span></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [5] jsonlite_1.7.2   magrittr_2.0.1   evaluate_0.14    highr_0.9       </span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [9] httr_1.4.2       stringi_1.7.3    curl_4.3.2       tools_4.1.0     </span></span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [13] stringr_1.4.0    Rlabkey_2.8.0    xfun_0.25        compiler_4.1.0</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
